### PR TITLE
icu76: init at 76.1rc

### PR DESCRIPTION
**Pull request by afh**
Preparation for upcoming ICU 76 release by adding this draft PR for ICU 76 RC ([Release Notes](https://github.com/unicode-org/icu/releases/tag/release-76-rc))

**N.B.:** In order to be able to build an ICU RC some minor changes to `make-icu.nix` were needed. These have been placed in a separate commit that can and maybe should be removed prior to the final ICU 76 PR, unless, of course, they provide merit.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by afh** [comment]
Result of `nixpkgs-review pr 345784` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>icu76</li>
    <li>icu76.dev</li>
  </ul>
</details>

File: pkgs/development/libraries/icu/make-icu.nix [hunk 1]
```diff
@@ -8,9 +8,13 @@ let
 
   pname = "icu4c";
 
+  release = if lib.hasSuffix "rc" version
+            then "${lib.replaceStrings [ "1" ] [ "" ] (lib.replaceStrings [ "." ] [ "-" ] version)}"
+            else "${lib.replaceStrings [ "." ] [ "-" ] version}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/icu/make-icu.nix:13]:
```suggestion
  release = lib.replaceStrings [ "1" ] [ "" ]
    (if lib.hasSuffix "rc" version then (lib.replaceStrings [ "." ] [ "-" ] version) else lib.id);
```

Comment (afh) [reply pkgs/development/libraries/icu/make-icu.nix:13]: Thanks, @SuperSandro2000, my intent was to keep this PR open until the final release is published and then remove the commit that allows to build rc, unless of course this seems useful for nixpkgs.
What are your thoughts on this?

Comment (7c6f434c) [reply pkgs/development/libraries/icu/make-icu.nix:13]: I think the question is what is the motivation to open the PR based on an RC. To get ofBorg builds? That sounds like a repeatable motivation, so maybe the -rc handling changes can be merged if you are interested in opening such RC-based PRs in the future.

Comment (afh) [reply pkgs/development/libraries/icu/make-icu.nix:13]: Correct, @7c6f434c, given that icu is such a "low-level" package and updating the default icu pckage is likely going to cause a mass-rebuild I wanted to get a head start on the upcoming version.

I'll have a look at previous rc releases to see whether the diverge in the same manner from the official releases.

Comment (afh) [reply pkgs/development/libraries/icu/make-icu.nix:13]: Except for the tags `icu4j-cldr-1-9-rc` and `release-64-2-rc` tags for icu release candidates do not contain a "minor" version (see below).
If folks see value in adding support for icu rc releases to the nixpkgs I'm happy to clean up this PR.

What are your thoughts on this @7c6f434c and @SuperSandro2000?

<details><summary>icu rc tags</summary>

```sh
curl https://api.github.com/repos/unicode-org/icu/git/refs/tags \
  | jq .[].ref \
  | sed -e 's@refs/tags/@@' \
  | sort -V \
  | ag 'rc$'
icu4j-cldr-1-9-rc
icu4j-release-49-rc
icu4j-release-50-rc
icu4j-release-51-rc
icu4j-release-52-rc
icu4j-release-53-rc
icu4j-release-54-rc
icu4j-release-55-rc
icu4j-release-56-rc
icu4j-release-57-rc
icu4j-release-58-rc
icu-release-49-rc
icu-release-50-rc
icu-release-51-rc
icu-release-52-rc
icu-release-53-rc
icu-release-54-rc
icu-release-55-rc
icu-release-56-rc
icu-release-57-rc
icu-release-58-rc
release-59-rc
release-60-rc
release-61-rc
release-62-rc
release-63-rc
release-64-2-rc
release-64-rc
release-65-rc
release-66-rc
release-67-rc
release-68-rc
release-69-rc
release-70-rc
release-71-rc
release-72-rc
release-73-rc
release-74-rc
release-75-rc
release-76-rc
```


</details>

**Comment by 7c6f434c** [comment]
I am fine with having a commented thing only used for testing RC versions (and with adding an RC version so it is easier to test stuff)

**Comment by afh** [comment]
What are your thoughts on the latest changes to this PR, @7c6f434c. Is this heading into the direction you had in mind?

**Comment by 7c6f434c** [comment]
Is the current state consistent? If the `rc`-removing code is needed for the testing to work, it should not be commented out while also adding an RC?

**Comment by afh** [comment]
Ah yes, apologies, @7c6f434c, this PR was missing the temporary changes to enable rc testing.

**Comment by 7c6f434c** [comment]
Well, if there are problems with specific complicated packages, let the fix PR be possible before the upstream release…

**Pull request merged**